spec-tests/src/test/java/org/candlepin/spec/pools/RefreshPoolsTest.java

```diff
@@ -0,0 +1,883 @@
+
+        // Verify the product exists in its initial state
+        ProductDTO fetchedProd = adminClient.ownerProducts().getProductByOwner(ownerKey, prod.getId());
+        compareProducts(prod, fetchedProd);
+
+        // Consume the pool multiple times so we have entitlements to revoke
+        for (int i = 0; i < sub.getQuantity(); i++) {
+            ConsumerDTO user = Consumers.random(owner, ConsumerTypes.System);
+            user.setUsername("admin");
+            user = adminClient.consumers().createConsumer(user, "admin",
+            owner.getKey(), null, true);
+            ApiClient consumerClient = ApiClients.trustedConsumer(user.getUuid());
+
+            JsonNode entNode = consumerClient.consumers().bindPool(user.getUuid(), pool.getId(), 1);
+            assertEquals(1, entNode.size());
+
+            user = adminClient.consumers().getConsumer(user.getUuid());
+            assertEquals(1, user.getEntitlementCount());
+        }
+
+        // Verify the entitlement count for this pool
+        List<EntitlementDTO> ents = adminClient.pools().getPoolEntitlements(pool.getId());
+        assertEquals(sub.getQuantity(), ents.size());
+
+        // Modify the subscription upstream
+        sub.setQuantity(1L);
+        adminClient.hosted().updateSubscription(sub.getId(), sub);
+
+        refreshPools(adminClient, owner.getKey());
+        pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+        pool = pools.get(0);
+        assertEquals(sub.getQuantity(), pool.getQuantity());
+
+        // Verify the entitlement count has changed
+        ents = adminClient.pools().getPoolEntitlements(pool.getId());
+        assertEquals(1, ents.size());
+    }
+
+    private AsyncJobStatusDTO refreshPools(ApiClient client, String ownerKey) {
+        AsyncJobStatusDTO job = adminClient.owners().refreshPools(ownerKey, true);
+        assertNotNull(job);
+        job = adminClient.jobs().waitForJob(job);
+        assertEquals("FINISHED", job.getState());
+
+        return job;
+    }
+
+    private void compareProducts(ProductDTO expected, ProductDTO actual) {
+        assertEquals(expected.getId(), actual.getId());
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.getMultiplier(), actual.getMultiplier());
+        assertEquals(expected.getAttributes(), actual.getAttributes());
+    }
+
+    private void compareBrandings(BrandingDTO expected, BrandingDTO actual) {
+        assertEquals(expected.getProductId(), actual.getProductId());
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.getType(), actual.getType());
+    }
+
+    private void compareBrandings(Collection<BrandingDTO> expected, Collection<BrandingDTO> actual) {
+        expected.stream().forEach(brand -> {
+            brand.setCreated(null);
+            brand.setUpdated(null);
+        });
+        actual.stream().forEach(brand -> {
```

Comment:
`Collection` (or more specifically, `Iterable`) also has a `forEach` method. It's not necessary to build a stream first.

Reply:
Also, while I'm thinking on it; since these are the generated DTOs, you could do it inline with method chaining:

```
expected.forEach(brand -> brand.created(null).updated(null));
actual.forEach(brand -> brand.created(null).updated(null));
```

Reply:
Updated